router/src/infer.rs

```diff
@@ -87,10 +87,14 @@ impl Infer {
             })?;
 
         // Validate request
-        let valid_request = self.validation.validate(request).await?;
+        let valid_request = self.validation.validate(request).await.map_err(|err| {
+            metrics::increment_counter!("tgi_request_failure", "err" => "validation");
+            tracing::error!("{err}");
+            err
+        })?;
 
         // MPSC channel to communicate with the background batching task
-        let (response_tx, response_rx) = mpsc::unbounded_channel();
+        let (response_tx, response_rx) = flume::unbounded();
```

Comment:
@OlivierDehaene presumably this is because you found flume to be faster than tokio mpsc?

Reply:
Not really. 
I added it because it simplifies the validation workers as flume is multi-producer multi-consumer and then I thought it was a bit weird to have flume for this and not for the other mpsc channels in the repo.